File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumLength = function (s) {
    const freq = Array(26).fill(0)
    let n = s.length;
    for (let i = 0; i < n; i++)
        freq[s.charCodeAt(i) - 97]++;

    let res = 0;
    for (let i = 0; i < n; i++) {
        let idx = s.charCodeAt(i) - 97;
        if (freq[idx] > 2)
            freq[idx] -= 2;
    }
    freq.forEach(x => {
        res += x;
    });
    return res;
};

File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.java
class Solution {
    public int minimumLength(String s) {
        int n = s.length();
        int[] f = new int[26];
        for (char ch : s.toCharArray())
            f[ch - 'a']++;

        int res = 0;
        for (int i = 0; i < n; i++) {
            if (f[s.charAt(i) - 'a'] > 2)
                f[s.charAt(i) - 'a'] -= 2;
        }
        for (int fr : f)
            res += fr;

        return res;
    }
}